pkg/privilege/privileges/cache.go

```diff
@@ -2062,10 +2062,10 @@ func (p *MySQLPrivilege) fetchSchemaPrivileges(users, hosts []string) (rows [][]
 }
 
 type columnStr = string
-type columnStrs = []columnStr
+type columnStrs = map[columnStr]struct{}
 type privOnColumns = map[mysql.PrivilegeType]columnStrs
 
-func privOnColumnsToString(p privOnColumns) string {
+func privOnColumnsToString(p privOnColumns, sqlMode mysql.SQLMode) string {
```

Comment:
We need to test the different escape behavior on different SQL mode.

Reply:
[`b629d2c` (#63728)](https://github.com/pingcap/tidb/pull/63728/commits/b629d2c88caae88712f87ad4eaf2593777b83c69)